### ISSUE TITLE
Server incompatibilities making requests with cookies from ktor-client

### ISSUE THREAD

**Issue by jpd236** [ktorio/ktor#389]
I encountered the following inconsistencies between how ktor+apache issues requests and how curl/postman issue the same requests, which caused the server to return an error instead of a valid response:

1. [HttpCookies](https://github.com/ktorio/ktor/blob/8c6dc2d4686e74096fe2109a0d5b6e8bb2af9eed/ktor-client/ktor-client-core/src/io/ktor/client/features/cookies/HttpCookies.kt#L63) sets multiple Cookie headers, one per value:

    ```kotlin
    cookies.forEach { _, cookie ->
       header(HttpHeaders.Cookie, renderSetCookieHeader(cookie))
    }
    ```

    This is invalid according to https://tools.ietf.org/html/rfc6265#section-5.4, which states:

    "When the user agent generates an HTTP request, the user agent MUST NOT attach more than one Cookie header field."

    Instead, the cookie keys/values need to be joined by ";" in a single Cookie header.

2. ktor includes an "$x-enc" snippet in the Cookie header, which is not part of the cookie standard.

3. The cookie value has a comma. Ktor refuses to encode this when using RAW format, which seemed reasonable enough, but curl happily passes it along.

4. Curl sets an "Accept: \*/\*" header that Ktor/Apache doesn't.

Only after resolving each of these inconsistencies was I able to successfully make a request.

**Comment by cy6erGn0m** [comment]
`$x-enc` need to be filtered out at engine stage. We just never faced issues because of it yet but this need to be fixed.

**Comment by e5l** [comment]
Done in `master`

**Comment by LeaveX007** [comment]
Hy guys,

now i have the same problem in the actual version of ktor. Please check it ou my post: https://stackoverflow.com/questions/79556230/set-cookie-is-wrong-wrong-x-enc

Br
Mile
